core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -375,9 +375,11 @@ class GroupCoordinator(val brokerId: Int,
       }
 
       if (eligibleGroups.nonEmpty) {
-        groupManager.cleanupGroupMetadata(None, eligibleGroups, Long.MaxValue)
+        val offsetsRemoved = groupManager.cleanupGroupMetadata(eligibleGroups, group => {
```

Comment:
I'm debating whether we should grab the group lock in these functions. I know the caller is holding the lock already when it invokes them, but I'm wondering if we should be more defensive in case we change the logic. At a minimum, we can document the fact that the function is invoked while holding the lock.

Reply:
Thanks for the quick feedback. Seems fair. I added some comments to clarify this matter. Please let me know if you think they're not adequate. Thanks.